presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowServer.java

```
@@ -0,0 +1,308 @@
+                        fields.add(field);
+                    }
+                }
+            }
+            else if (selectStatement != null) {
+                selectStatement = selectStatement.toUpperCase();
+                logger.info("Executing SELECT query: " + selectStatement);
+                try (ResultSet rs = connection.createStatement().executeQuery(selectStatement)) {
+                    ResultSetMetaData metaData = rs.getMetaData();
+                    int columnCount = metaData.getColumnCount();
+
+                    for (int i = 1; i <= columnCount; i++) {
+                        String columnName = metaData.getColumnName(i);
+                        String columnType = metaData.getColumnTypeName(i);
+                        int precision = metaData.getPrecision(i);
+                        int scale = metaData.getScale(i);
+
+                        ArrowType arrowType = convertSqlTypeToArrowType(columnType, precision, scale);
+                        Field field = new Field(columnName, FieldType.nullable(arrowType), null);
+                        fields.add(field);
+                    }
+                }
+            }
+            else {
+                throw new IllegalArgumentException("Either schema_name/table_name or select_statement must be provided.");
+            }
+
+            Schema schema = new Schema(fields);
+            FlightEndpoint endpoint = new FlightEndpoint(new Ticket(flightDescriptor.getCommand()));
+            return new FlightInfo(schema, flightDescriptor, Collections.singletonList(endpoint), -1, -1);
+        }
+        catch (Exception e) {
+            logger.error(e);
+            throw new RuntimeException("Failed to retrieve FlightInfo", e);
+        }
+    }
+
+    @Override
+    public Runnable acceptPut(CallContext callContext, FlightStream flightStream, StreamListener<PutResult> streamListener)
+    {
+        return null;
```

Please throw UnsupportedOperationException

---

Updated